### PR TITLE
Nodejs generated unit tests

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

**Comment by andreamlin** [comment]
This is built on https://github.com/googleapis/gapic-generator/pull/2283, which is a fix for node. Let that get merged in first.

**Comment by andreamlin** [comment]
PTAL

File: .circleci/config.yml [hunk 1]
```diff
@@ -235,6 +235,16 @@ anchor_run_php: &anchor_run_php
         name: Test PHP Logging client
         <<: *anchor_test_php
 
+# Test a NodeJS client if generation succeeded.
+anchor_test_nodejs_client: &anchor_test_nodejs_client
+  command: |
+    cd ${NODEJS_GEN_DIR}/${TEST_API}
+    npm install
+    # Copy over IAM protos.
+    cp -r node_modules/google-proto-files/google/iam/ protos/google/
```
Comment (alexander-fenster) [inline .circleci/config.yml]: why do we need to copy those? In other words, why we don't need to copy those for artman-generated client libraries? Does artman do some extra steps that gapic-generator does not?

Comment (andreamlin) [reply .circleci/config.yml]: `${NODEJS_GEN_DIR}/${TEST_API}` is an artman generated library. I don't know why artman doesn't place the necessary protos in the protos/google directory of the generated lib.

Comment (andreamlin) [reply .circleci/config.yml]: Looking at the existing [nodejs-pubsub/proto](https://github.com/googleapis/nodejs-pubsub/tree/master/protos/google) dir, those protos haven't been updated for a long time, so they've just always been there, so node automation doesn't have to deal with copying over the protos

Comment (alexander-fenster) [reply .circleci/config.yml]: Well, it makes sense. OK let's leave this copy here for now, even though I feel it's not exactly correct :)

**Pull request merged**